### PR TITLE
`merkledb` -- use `Maybe` for start bounds

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

We already use `Maybe` for end bounds so we should use them for start bounds for xonsistency.

## How this works

Change `[]byte` to `maybe.Maybe[[]byte]` for start bounds.

## How this was tested

Update existing UT

### CODE REVIEW

File: x/sync/workheap.go [hunk 1]
```diff
@@ -98,13 +111,15 @@ func (wh *workHeap) MergeInsert(item *workItem) {
...
 		func(beforeItem *heapItem) bool {
-			if item.localRootID == beforeItem.workItem.localRootID && bytes.Equal(beforeItem.workItem.end.Value(), item.start) {
-				// [beforeItem.start, beforeItem.end] and [item.start, item.end] are
-				// merged into [beforeItem.start, item.end]
-				beforeItem.workItem.end = item.end
-				beforeItem.workItem.priority = math.Max(item.priority, beforeItem.workItem.priority)
-				heap.Fix(&wh.innerHeap, beforeItem.heapIndex)
-				mergedBefore = beforeItem
+			if item.localRootID == beforeItem.workItem.localRootID {
+				if maybe.Equal(item.start, beforeItem.workItem.end, bytes.Equal) {
```
Comment (dboehm-avalabs) [inline x/sync/workheap.go:115]: reason to split the if?

File: x/sync/workheap.go [hunk 2]
```diff
@@ -114,13 +129,15 @@ func (wh *workHeap) MergeInsert(item *workItem) {
...
 		func(afterItem *heapItem) bool {
-			if item.localRootID == afterItem.workItem.localRootID && bytes.Equal(afterItem.workItem.start, item.end.Value()) {
-				// [item.start, item.end] and [afterItem.start, afterItem.end] are merged into
-				// [item.start, afterItem.end].
-				afterItem.workItem.start = item.start
-				afterItem.workItem.priority = math.Max(item.priority, afterItem.workItem.priority)
-				heap.Fix(&wh.innerHeap, afterItem.heapIndex)
-				mergedAfter = afterItem
+			if item.localRootID == afterItem.workItem.localRootID {
+				if maybe.Equal(item.end, afterItem.workItem.start, bytes.Equal) {
```
Comment (dboehm-avalabs) [inline x/sync/workheap.go:133]: reason to split the if?

**Review by dboehm-avalabs — APPROVED** [review]

File: utils/maybe/maybe.go [hunk 3]
```diff
@@ -42,6 +47,13 @@ func (m Maybe[T]) Value() T {
 	return m.value
 }
 
+func (m Maybe[T]) String() string {
```
Comment (dboehm-avalabs) [inline utils/maybe/maybe.go:50]: was this used anywhere?

Comment (danlaine) [reply utils/maybe/maybe.go:50]: Yes in `manager.go`:

```go
		zap.Stringer("start", work.start),
		zap.Stringer("end", largestHandledKey),
```

**Review by dboehm-avalabs — APPROVED** [review]

**Pull request merged**